src/lib/InstantSearch.ts

```diff
@@ -245,23 +246,22 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
   }
 
   /**
-   * Adds a widget. This can be done before and after InstantSearch has been started. Adding a
-   * widget after InstantSearch started is considered **EXPERIMENTAL** and therefore
-   * it is possibly buggy, if you find anything please
-   * [open an issue](https://github.com/algolia/instantsearch.js/issues/new?title=Problem%20with%20hot%20addWidget).
-   * @param  widget The widget to add to InstantSearch. Widgets are simple objects
-   * that have methods that map the search life cycle in a UI perspective. Usually widgets are
-   * created by [widget factories](widgets.html) like the one provided with InstantSearch.js.
+   * Adds a widget. This can be done before and after InstantSearch has been started.
+   * @param widget The widget to add to InstantSearch.
+   *
+   * This method is deprecated, and `addWidget(widget)` can be replaced by `addWidgets([widget])`
    */
   public addWidget(widget: Widget) {
+    warning(
+      false,
+      'addWidget(widget) is replaced by addWidgets([widgets]) to help see the structure'
```

Comment:
```suggestion
      '`addWidget(widget)` is deprecated, prefer the use of `addWidgets([widgets])`'
```

Would be nicer for the user. Or something similar.

Reply:
```suggestion
      'This method will still be supported in 4.x releases, but not further. It is replaced by `addWidgets([widget])`'
```